src/providers/api-interceptor/api-interceptor.spec.ts

```diff
@@ -0,0 +1,166 @@
+// Mocha global variables (for Windows)
+/// <reference path="../../../node_modules/@types/mocha/index.d.ts" />
+
+import { HttpClient, HttpHeaders, HTTP_INTERCEPTORS, HttpParams, HttpRequest } from '@angular/common/http';
+import { TestBed } from '@angular/core/testing';
+import { HttpClientTestingModule, HttpTestingController } from '@angular/common/http/testing';
+import { expect } from 'chai';
+import { compact, isEqual } from 'lodash';
+import { Observable } from 'rxjs/Rx';
+
+import { HeadersOrParams, httpRequestMatcher } from '../../../spec/http';
+import EnvService from '../env-service/env-service';
+import { ApiInterceptor } from './api-interceptor';
+
+type EnvServiceMock = Partial<EnvService>;
+
+describe('ApiInterceptor', function () {
+  let envServiceMock: EnvServiceMock;
+  let httpClient: HttpClient;
+  let httpTestingCtrl: HttpTestingController;
+
+  beforeEach(function () {
+    envServiceMock = {
+      environment: 'development',
+      backendUrl: 'https://example.com/api'
+    };
+
+    TestBed.configureTestingModule({
+      imports: [
+        HttpClientTestingModule
+      ],
+      providers: [
+        { provide: EnvService, useValue: envServiceMock },
+        // Plug the interceptor in so that we can make HTTP requests
+        // and see if it does its job correctly.
+        {
+          provide: HTTP_INTERCEPTORS,
+          useClass: ApiInterceptor,
+          multi: true
+        }
+      ]
+    });
+
+    httpClient = TestBed.get(HttpClient);
+    httpTestingCtrl = TestBed.get(HttpTestingController);
+  });
+
+  afterEach(function() {
+    // Make sure no extra HTTP requests were made.
+    httpTestingCtrl.verify();
+  });
+
+  type TestDefinition = {
+    method: string;
+    body?: any;
+    options?: {
+      headers?: HeadersOrParams;
+      params?: HeadersOrParams;
+    }
+  };
+
+  // This array various defines HTTP request parameters to use for the following tests.
```

Comment:
Switched the word "various" and "defines", I suppose ?
Should be : "defines various" instead